WordPress/gutenberg
# Footnotes: try with post meta

## What?


Previously: #50124, #49797, #47682, #28261, #15267.

Similar to #50124, but storing it with post meta.

Many approaches have been explored, including storing footnotes inline and in the footnotes block, but eventually we decided to go for storing them in post meta to make it easier to render the footnotes outside of the post content in templates.

To do: fix batching (#51644)

## Why?


## How?


This PR implements footnotes by:

* Registering a custom meta field where we store the footnotes in the correct order.
* The order is maintained by the entity provider, which looks through all rich text values in order, rearranges the footnotes array when necessary, but also adds and removes footnotes when they appear in the content or disappear.
* A footnotes block is inserted at the bottom when the first footnote is added. This block simply renders the footnotes it finds in the meta field, nothing more. This blocks could be moved anywhere, probably also outside of the post content.
* There's a rich text format that adds a button to insert footnote anchors. It's also responsible for rendering them inside rich text, making sure it's not editable, selectable etc.

## Testing Instructions

Create a post, add a paragraph and click the "more" section in the rich text section to insert a new footnote anchor. A footnotes block should appear at the bottom where you can add the note. Add a few in different places, rearrange them, and check that the order at the bottom remains correct. Also test copy pasting.

### Testing Instructions for Keyboard


## Screenshots or screencast
